README.md

```diff
@@ -64,13 +60,46 @@ For the full reference of possible configuration values, see the [config schema
     make help
     ```
 
-    > Note: If you're deploying for the first time and not using `make all` (i.e. using `make deploy-core`), ensure you have ran `make bootstrap` first.
-
 ### CI (GitHub Actions)
 
 CI deployment workflows are run in [Github environments](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment). These should
 be created in a private repository created from this template repository.
 
-<!-- 
-very much a work in progress here...
--->
+1. Create a service principal
+
+    CI deployments require a service principal with access to deploy resources
+    in the subscription. Create one with:
```

Comment:
It would not make sense to use a service principal that is shared among all, i.e. no need to create one?

Reply:
all what?..
--
think one needs to be created per subscription, which likely means per GH environment

Reply:
Among all GH environments if there is ever multiple, which I also don't know